### PR TITLE
fix error during mkcol

### PR DESCRIPTION
**Pull request by micbar**
# Description

We need to return `409 Conflict` when one or more path elements during an MKCOL do not exist.

Fixes: https://github.com/owncloud/ocis/issues/6136

### CODE REVIEW

**Comment by micbar** [comment]
@dragotin This is a fix to enable the usage of `rclone`

**Comment by glpatcern** [comment]
@redblom do we need this in `master` or do we already have this behaviour?

**Comment by redblom** [comment]
@micbar @glpatcern : I can confirm that `master` returns `409 Conflict` and rclone behaves as expected, it can create a complete new path.

**Comment by micbar** [comment]
> @micbar @glpatcern : I can confirm that `master` returns `409 Conflict` and rclone behaves as expected, it can create a complete new path.

@redblom Can you check if this is also the case for non-existing sub-sub directories?

**Review by butonic — CHANGES_REQUESTED** [review]

File: pkg/storage/utils/decomposedfs/decomposedfs.go [hunk 1]
```diff
@@ -553,6 +553,9 @@ func (fs *Decomposedfs) CreateDir(ctx context.Context, ref *provider.Reference)
 	// verify parent exists
 	var n *node.Node
 	if n, err = fs.lu.NodeFromResource(ctx, parentRef); err != nil {
```
[the remark this answers is not in the document]
Comment (micbar) [reply pkg/storage/utils/decomposedfs/decomposedfs.go:555]: Can you check why this is not working as you expect it? The problem is, that this only works on the first level.

**Comment by redblom** [comment]
> @redblom Can you check if this is also the case for non-existing sub-sub directories?

@micbar Yes, also with sub/sub/sub directories, the response is always `409 Conflict` (message: `Parent node does not exist`)

File: pkg/storage/utils/decomposedfs/decomposedfs.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (micbar) [reply pkg/storage/utils/decomposedfs/decomposedfs.go]: done

**Review by butonic — APPROVED** [review]

**Pull request merged**